packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -37,12 +39,32 @@ export interface BucketDeploymentProps {
    * @default true - when resource is deleted/updated, files are retained
    */
   readonly retainOnDelete?: boolean;
+
+  /**
+   * The CloudFront distribution using the destination bucket as an origin.
+   * Files in the distribution's edge caches will be invalidated after
+   * files are uploaded to the destination bucket.
+   *
+   * @default No invalidation occurs
+   */
+  readonly distribution?: cloudfront.IDistribution;
+
+  /**
+   * The file paths to invalidate in the CloudFront distribution.
+   *
+   * @default All files under the destination bucket key prefix will be invalidated.
+   */
+  readonly distributionPaths?: string[];
 }
 
 export class BucketDeployment extends cdk.Construct {
   constructor(scope: cdk.Construct, id: string, props: BucketDeploymentProps) {
     super(scope, id);
 
+    if (props.distributionPaths && !props.distribution) {
+      throw new Error("Distribution must be specified if distribution paths are specified");
+    }
+
     const handler = new lambda.SingletonFunction(this, 'CustomResourceHandler', {
```

Comment:
Would be nice to be able to validate that the cloudfront distribution has the bucket as a source origin.
Can you please raise an issue?

Reply:
It might be tricky to find all possible and valid ways. You can add S3 in many different ways. What issue should be raised and where?

Reply:
Just raise an issue that describes the desire to be able to check that a web distribution has a specific bucket origin.